docs/flatcar-linux/kvm-libvirt.md

```diff
@@ -0,0 +1,297 @@
+# KVM libvirt
+
+In this tutorial, we'll create a Kubernetes cluster with Flatcar Linux inside VMs on our local machine.
+
+We'll declare the cluster using the Lokomotive Terraform module for the KVM libvirt platform.
+
+Controllers are provisioned to run an `etcd-member` peer and a `kubelet` service. Workers run just a `kubelet` service. A one-time [bootkube](https://github.com/kubernetes-incubator/bootkube) bootstrap schedules the `apiserver`, `scheduler`, `controller-manager`, and `coredns` on controllers and schedules `kube-proxy` and `calico` (or `flannel`) on every node. A generated `kubeconfig` provides `kubectl` access to the cluster.
+
+## Requirements
+
+* free RAM (minimum 4 GB)
+* libvirtd and a user who is allowed to create virtual machines
+* Terraform v0.11.x, [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct), and [terraform-provider-libvirt](https://github.com/dmacvicar/terraform-provider-libvirt) installed locally
+
+## libvirtd and User Setup
+
+*Installation on your platform:* Since we will build the terraform provider, not only libvirt should be installed but also the development packets.
+For example in Fedora it would be `sudo dnf install libvirt-daemon libvirt-devel`.
+
+Ensure that libvirt is set up and accessible for the current user:
+
+```sh
+$ sudo systemctl start libvirtd
+$ sudo usermod -a -G libvirt $(whoami)
+$ newgrp libvirt
+```
+
+## Terraform Setup
+
+Install [Terraform](https://www.terraform.io/downloads.html) v0.11.x on your system.
+
+```sh
+$ terraform version
+Terraform v0.11.13
+```
+
+Add the [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct) plugin binary for your system to `~/.terraform.d/plugins/`, noting the final name.
+
+```sh
+wget https://github.com/poseidon/terraform-provider-ct/releases/download/v0.3.1/terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+tar xzf terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+mv terraform-provider-ct-v0.3.1-linux-amd64/terraform-provider-ct ~/.terraform.d/plugins/terraform-provider-ct_v0.3.1
+```
+
+Build terraform-provider-libvirt from source because pool definition feature was not released yet:
+
+```sh
+$ go get github.com/dmacvicar/terraform-provider-libvirt
+$ cd $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt
+$ GO111MODULE=on GOFLAGS=-mod=vendor make install
+$ cp $GOPATH/bin/terraform-provider-libvirt ~/.terraform.d/plugins/terraform-provider-libvirt
+```
+
+Read [concepts](/docs/architecture/concepts.md) to learn about Terraform, modules, and organizing resources. Change to your infrastructure repository (e.g. `infra`).
+
+```sh
+cd infra/vmcluster
+```
+
+## Provider
+
+Create the `provider.tf` file:
```

Comment:
I suggest "Create a file named `provider.tf`". The current phrasing assumes the user knows about this file.